pynamodb/attributes.py

```diff
@@ -972,6 +972,59 @@ def as_dict(self):
         return result
 
 
+class DynamicMapAttribute(MapAttribute):
+    """
+    A map attribute that supports declaring attributes (like an AttributeContainer) but will also store
+    any other values that are set on it (like a raw MapAttribute).
+
+    >>> class MyDynamicMapAttribute(DynamicMapAttribute):
+    >>>     a_date_time = UTCDateTimeAttribute()  # raw map attributes cannot serialize/deserialize datetime values
+    >>>
+    >>> dynamic_map = MyDynamicMapAttribute()
+    >>> dynamic_map.a_date_time = datetime.utcnow()
+    >>> dynamic_map.a_number = 5
+    >>> dynamic_map.serialize()  # {'a_date_time': {'S': 'xxx'}, 'a_number': {'N': '5'}}
+    """
+
+    def __setattr__(self, name, value):
+        # Set attributes via the Attribute descriptor if it exists.
+        object.__setattr__(self, name, value) if name in self.get_attributes() else super().__setattr__(name, value)
+
+    def serialize(self, values):
+        if not isinstance(values, type(self)):
```

Comment:
this blob transforms a raw dict into a map attr object right? is this not shared with raw map attrs?